src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -431,8 +431,22 @@ private sealed class BufferMemoryReceiveOperation : ReceiveOperation
 
             public BufferMemoryReceiveOperation(SocketAsyncContext context) : base(context) { }
 
-            protected override bool DoTryComplete(SocketAsyncContext context) =>
-                SocketPal.TryCompleteReceiveFrom(context._socket, Buffer.Span, null, Flags, SocketAddress, ref SocketAddressLen, out BytesTransferred, out ReceivedFlags, out ErrorCode);
+            protected override bool DoTryComplete(SocketAsyncContext context)
+            {
+                // Zero byte read is performed to know when data is available.
+                // We don't have to call receive, our caller is interested in the event.
```

Comment:
Are there any concerns about the state in which this leaves the epoll, or anything else that might be left behind by not "cleaning up" by issuing the receive?

Reply:
No. epoll isn't aware we're doing a receive. It knows only we are interested in readable events for this file descriptor.